src/main/java/runhistoryplus/patches/ImprovableCardRunHistoryPatch.java

```diff
@@ -0,0 +1,88 @@
+package runhistoryplus.patches;
+
+import basemod.ReflectionHacks;
+import com.evacipated.cardcrawl.modthespire.lib.*;
+import com.evacipated.cardcrawl.modthespire.patcher.PatchingException;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.cards.CardGroup;
+import com.megacrit.cardcrawl.cards.blue.GeneticAlgorithm;
+import com.megacrit.cardcrawl.cards.colorless.RitualDagger;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.metrics.Metrics;
+import com.megacrit.cardcrawl.monsters.MonsterGroup;
+import com.megacrit.cardcrawl.screens.runHistory.RunHistoryScreen;
+import com.megacrit.cardcrawl.screens.runHistory.TinyCard;
+import com.megacrit.cardcrawl.screens.stats.RunData;
+import javassist.CannotCompileException;
+import javassist.NotFoundException;
+import javassist.CtBehavior;
+import javassist.CtClass;
+import javassist.CtField;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ImprovableCardRunHistoryPatch {
+    private static Map<String, List<Integer>> improvableCardsLog;
+    private static final ArrayList<String> improvableCards = new ArrayList<String>() {{
+        add(GeneticAlgorithm.ID);
+        add(RitualDagger.ID);
+    }};
+
+    @SpirePatch(clz = CardCrawlGame.class, method = SpirePatch.CONSTRUCTOR)
+    public static class ImprovableCardsField {
+        @SpireRawPatch
+        public static void addImprovableCardsField(CtBehavior ctBehavior) throws NotFoundException, CannotCompileException {
+            CtClass runData = ctBehavior.getDeclaringClass().getClassPool().get(RunData.class.getName());
+            String fieldSource = "public java.util.Map improvable_cards;";
+            CtField field = CtField.make(fieldSource, runData);
+            runData.addField(field);
+        }
+    }
+
+    @SpirePatch(clz = Metrics.class, method = "gatherAllData")
+    public static class GatherAllDataPatch {
+        @SpirePostfixPatch
+        public static void gatherAllDataPatch(Metrics __instance, boolean death, boolean trueVictor, MonsterGroup monsters) {
+            improvableCardsLog = new HashMap<>();
+            for (final AbstractCard card: AbstractDungeon.player.masterDeck.group) {
+                if (improvableCards.contains(card.cardID)) {
+                    if (!improvableCardsLog.containsKey(card.name)) {
+                        improvableCardsLog.put(card.name, new ArrayList<>());
+                    }
+                    improvableCardsLog.get(card.name).add(card.misc);
```

Comment:
This makes sense in terms of what we're storing -- the list of values for each version of the card.

I completely agree with storing the full list of values (instead of just the highest one). It means that anyone doing analysis on the log files has access to all of it, even if the run history screen only shows the highest value. And it means that if in the future we come up with a good way to show all the values, all we need to do is change the display code.